bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/protocol/MPDConnectionThread.java

```diff
@@ -0,0 +1,298 @@
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.mpd.internal.MPDException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class for communicating with the music player daemon through a IP connection
+ *
+ * @author Stefan Röllin - Initial contribution
+ */
+@NonNullByDefault
+public class MPDConnectionThread extends Thread {
+
+    private static final int RECONNECTION_TIMEOUT_SEC = 60;
+
+    private final Logger logger = LoggerFactory.getLogger(MPDConnectionThread.class);
+
+    private final MPDResponseListener listener;
+
+    private final String address;
+    private final Integer port;
+    private final String password;
+
+    private @Nullable Socket socket = null;
+    private @Nullable InputStreamReader inputStreamReader = null;
+    private @Nullable BufferedReader reader = null;
+    private @Nullable DataOutputStream writer = null;
+
+    private final List<MPDCommand> pendingCommands = new ArrayList<>();
+    private AtomicBoolean isInIdle = new AtomicBoolean(false);
+    private AtomicBoolean disposed = new AtomicBoolean(false);
+
+    public MPDConnectionThread(MPDResponseListener listener, String address, Integer port, String password) {
+        super(MPDConnectionThread.class.getSimpleName() + " " + address + ":" + port);
+
+        this.listener = listener;
+
+        this.address = address;
+        this.port = port;
+        this.password = password;
+        setDaemon(true);
+    }
+
+    @Override
+    public void run() {
+        try {
+            while (!disposed.get()) {
+                try {
+                    synchronized (pendingCommands) {
+                        pendingCommands.clear();
+                        pendingCommands.add(new MPDCommand("status"));
+                        pendingCommands.add(new MPDCommand("currentsong"));
+                    }
+
+                    establishConnection();
+                    updateThingStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, null);
+
+                    processPendingCommands();
+                } catch (IOException e) {
+                    updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+                } catch (MPDException e) {
+                    updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+                }
+
+                isInIdle.set(false);
+                closeSocket();
+
+                if (!disposed.get()) {
+                    sleep(RECONNECTION_TIMEOUT_SEC * 1000);
+                }
+            }
+        } catch (InterruptedException e) {
```

Comment:
You could consider renaming `e` to `ignore` this makes the ignoring of the exception explicit and documented.

```suggestion
        } catch (InterruptedException ignore) {
```